### PR TITLE
Add backward pass support to addmm and gemm operators

### PR DESCRIPTION
**Pull request by tianrengao**
- Add get_bwd_fn method to both operators 
- Add requires_grad detection for BWD and FWD_BWD modes in addmm operator
- Complete existing partial backward support in gemm operator
- Enable tritonbench backward accuracy testing for addmm-bwd and gemm-bwd

### CODE REVIEW

**Comment by xuzhao9** [comment]
Can you keep the files changed only related to this PR?

**Comment by tianrengao** [comment]
> Can you keep the files changed only related to this PR?

Sure it was an accident. Done, could you review again, thanks!
